sympy/physics/mechanics/tests/test_jointsmethod.py

```diff
@@ -0,0 +1,89 @@
+from sympy import symbols, Matrix, cos, sin, expand
+from sympy.physics.mechanics import (PinJoint, JointsMethod, Body, KanesMethod,
+                                    PrismaticJoint, LagrangesMethod)
+from sympy.physics.vector import dynamicsymbols, ReferenceFrame
+
+
+t = dynamicsymbols._t
+
+
+def test_jointsmethod():
+    P = Body('P')
+    C = Body('C')
+    Pin = PinJoint('P1', P, C)
+    C_ixx, g = symbols('C_ixx g')
+    theta, omega = dynamicsymbols('theta_P1, omega_P1')
+    P.apply_force(g*P.y)
+    method = JointsMethod(P, Pin)
+    assert method.frame == P.frame
+    assert method.bodies == [C, P]
+    assert method.loads == [(P.masscenter, g*P.frame.y)]
+    assert method.q == [theta]
+    assert method.u == [omega]
+    assert method.kdes == [omega - theta.diff()]
+    soln = method.form_eoms()
+    assert soln == Matrix([[-C_ixx*omega.diff()]])
+    assert method.forcing_full == Matrix([[omega], [0]])
+    assert method.mass_matrix_full == Matrix([[1, 0], [0, C_ixx]])
+    assert isinstance(method.method, KanesMethod)
+
+def test_complete_simple_double_pendulum():
+    q1, q2 = dynamicsymbols('q1 q2')
+    u1, u2 = dynamicsymbols('u1 u2')
+    m, l = symbols('m l')
+    N = ReferenceFrame('N')
+    A = ReferenceFrame('A')
+    B = ReferenceFrame('B')
+    C = Body('C', frame=N)  # ceiling
+    PartP = Body('P', frame=A, mass=m)
+    PartR = Body('R', frame=B, mass=m)
+
+    J1 = PinJoint('J1', C, PartP, speeds=u1, coordinates=q1,
+                  child_joint_pos=-l*A.x, parent_axis=C.frame.z,
+                  child_axis=PartP.frame.z)
```

Comment:
```suggestion
                  child_axis=PartP.z)
```

Reply:
Change all to use the Body attributes if possible.

Reply:
Done!